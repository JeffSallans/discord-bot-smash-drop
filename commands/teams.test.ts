import * as seed from 'seed-random';
seed('123', { global: true });

let importedCommands, importedReadyForCommands, destroy;

describe('teams', () => {
  let mockMessageObject;

  beforeAll(() => {
    const { commands, readyForCommands, destroy } = require('..');
    importedCommands = commands;
    importedReadyForCommands = readyForCommands;
    mockMessageObject = {};
    mockMessageObject.reply = (message) => {
      expect(message).toMatchSnapshot();
    }
  });

  afterAll(async () => {
    await destroy();
  });

  it('works with no players', async () => {
    await importedReadyForCommands;
    await importedCommands['teams'](mockMessageObject, ['teams']);
  });

  it('works with 4 players', async () => {
    await importedReadyForCommands;
    await importedCommands['teams'](mockMessageObject, ['teams', 'math', 'mlripper', 'thynix', 'hasp']);
  });

  it('works with uneven players', async () => {
    await importedReadyForCommands;
    await importedCommands['teams'](mockMessageObject, ['teams', 'math', 'mlripper', 'thynix', 'hasp', 'qazwode']);
  });

  it('works with 6 players', async () => {
    await importedReadyForCommands;
    await importedCommands['teams'](mockMessageObject, ['teams', 'math', 'mlripper', 'thynix', 'hasp', 'qazwode', 'csg']);
  });
});
