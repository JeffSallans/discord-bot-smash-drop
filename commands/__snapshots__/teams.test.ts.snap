// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`teams works with 4 players 1`] = `
"Left Team: hasp, thynix
Right Team: math, mlripper"
`;

exports[`teams works with 6 players 1`] = `
"Left Team: math, mlripper, csg
Right Team: qazwode, hasp, thynix"
`;

exports[`teams works with no players 1`] = `
"Left Team:
Right Team: "
`;

exports[`teams works with uneven players 1`] = `
"Left Team: qazwode, hasp
Right Team: mlripper, math, thynix"
`;
